jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudformation/json/JsonCfnParserTest.kt

```diff
@@ -0,0 +1,162 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudformation.json
+
+import com.intellij.json.psi.JsonFile
+import com.intellij.psi.PsiFile
+import com.intellij.testFramework.runInEdtAndWait
+import org.apache.commons.lang3.builder.MultilineRecursiveToStringStyle
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.fail
+import org.junit.Rule
+import org.junit.Test
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnParsedFile
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnProblem
+import software.aws.toolkits.jetbrains.utils.rules.JavaCodeInsightTestFixtureRule
+import kotlin.reflect.full.isSubclassOf
+import kotlin.test.assertNotNull
+
+class JsonCfnParserTest {
+    @Rule
+    @JvmField
+    val projectRule = JavaCodeInsightTestFixtureRule()
```

Comment:
Could we do parameterized tests instead? (just read about  junit `@Parameters` which should work for this)

Reply:
Ya, will change it in next revision. Need to fix the build failing on 193 too